src/briefcase/integrations/java.py

```diff
@@ -0,0 +1,201 @@
+                return Path(java_home)
+            else:
+                # It's not a Java 8 JDK.
+                print("""
+Android requires a Java 8 JDK, but the location pointed to by JAVA_HOME
+isn't a Java 8 JDK (it appears to be Java {version_str}).
+
+Briefcase will use it's own JDK instance.
+""".format(version_str=version_str))
+
+        except FileNotFoundError:
+            print("""
+The location pointed to by JAVA HOME does not appear to be a JDK.
+It may be a Java Runtime Environment (JRE).
+
+Briefcase will use it's own JDK instance.
+""")
+
+        except subprocess.CalledProcessError:
+            print("""
+*************************************************************************
+** WARNING: Unable to invoke the Java compiler                         **
+*************************************************************************
+
+   Briefcase received an unexpected error when trying to invoke javac,
+   the Java compiler, at the location indicated by JAVA_HOME.
+
+   Briefcase will continue by downloading and using it's own JDK.
+
+   Please report this as a bug at:
+
+     https://github.com/beeware/briefcase/issues/
+
+
+   In your report, please including the output from running:
+
+     {java_home}/bin/javac -version
+
+   from the command prompt.
+
+*************************************************************************
+
+""".format(java_home=java_home))
+
+        except IndexError:
+            print("""
+*************************************************************************
+** WARNING: Unable to determine the version of Java that is installed  **
+*************************************************************************
+
+   Briefcase was unable to interpret the version information returned
+   by the Java compiler at the location indicated by JAVA_HOME.
+
+   Briefcase will continue by downloading and using it's own JDK.
+
+   Please report this as a bug at:
+
+     https://github.com/beeware/briefcase/issues/
+
+
+   In your report, please including the output from running:
+
+     {java_home}/bin/javac -version
+
+   from the command prompt.
+
+*************************************************************************
+
+""".format(java_home=java_home))
+
+    java_home = cmd.dot_briefcase_path / 'tools' / 'java'
+
+    # The macOS download has a weird layout (inherited from the official Oracle
+    # release). The actual JAVA_HOME is deeper inside the directory structure.
+    if cmd.host_os == 'Darwin':
+        java_home = java_home / 'Contents' / 'Home'
+
+    if (java_home / 'bin').exists():
+        # Using briefcase cached Java version
+        # print("Using a Java 8 JDK cached by Briefcase...")
+        return java_home
+
+    print("Obtaining a Java 8 JDK...")
+
+    # As of April 10 2020, 8u242this is the current AdoptOpenJDK
+    # https://adoptopenjdk.net/releases.html
+    jdk_release = '8u242'
+    jdk_build = 'b08'
```

Comment:
I did this more for brevity; the URL is *loooooooong*, and this was the best way to avoid having three absurdly long URL names, when we *also* need to track the name of the release + build for the rename.